File: ddd/src/main/java/yun/blog/ddd/order/Order.java
package yun.blog.ddd.order;

import java.util.ArrayList;
import java.util.List;
import javax.persistence.Embedded;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.Table;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;
import yun.blog.ddd.model.Address;
import yun.blog.ddd.product.Product;

@Entity
@Table(name = "orders")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Getter
public class Order {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  @Embedded
  private Address address;

  @OneToMany(mappedBy = "order")
  private List<Product> products = new ArrayList<>();


}

File: yun/domain/src/main/kotlin/com/cheese/yun/domain/order/Order.kt
package com.cheese.yun.domain.order

import com.cheese.yun.domain.model.Address
import com.cheese.yun.domain.support.EntityAuditing
import java.math.BigDecimal
import javax.persistence.*

@Entity
@Table(name = "orders")
class Order(

    @Embedded
    @AttributeOverrides(
        AttributeOverride(name = "zipCode", column = Column(name = "zip_code", nullable = false)),
        AttributeOverride(name = "city", column = Column(name = "city", nullable = false)),
        AttributeOverride(name = "detailAddress", column = Column(name = "detail_address", nullable = false))
    )
    val address: Address,

    @Column(name = "price", nullable = false)
    val price: BigDecimal
) : EntityAuditing() {


}